nucypher/cli/characters/ursula.py

```diff
@@ -162,9 +165,23 @@ def ursula(click_config,
         if not config_root:                         # Flag
             config_root = click_config.config_file  # Envvar
 
+        # Attempts to automatically get the external IP from ifconfig.me
+        # If the request fails, it falls back to the standard process.
         if not rest_host:
-            # TODO: Remove this step - use a fleet node to detect the IP
-            rest_host = click.prompt("Enter Ursula's public-facing IPv4 address")
+            rest_host = actions.get_external_ip()
+            is_valid_address = False
+            if rest_host is not None and not force:
+                is_valid_address = click.confirm(f"Is this the public-facing IPv4 address ({rest_host}) you want to use for Ursula?")
+            if not is_valid_address or rest_host is not None:
```

Comment:
If `rest_host` is an actual IP address and `force` is true -> this branch is enterred and a RuntimeError is thrown.

Reply:
Nice catch.

Reply:
@tuxxy what about something like:
```python
    rest_host = actions.get_external_ip()
    if rest_host is None and force:
        # could not determine ip automatically but force is set
        raise RuntimeError("There was an error automatically determining the IP address")
    else:
        is_valid_address = rest_host is not None and force  # assume valid since successfully determined and force set to true
        if rest_host is not None and not force:
            # confirm validity since not forced
            is_valid_address = click.confirm(f"Is this the public-facing IPv4 address ({rest_host}) yo\
u want to use for Ursula?")

        if not is_valid_address:
            # unable to determine valid ip, prompt user for it
            rest_host = click.prompt("Please enter Ursula's public-facing IPv4 address here:")
```